components/renku_data_services/crc/blueprints.py

```diff
@@ -203,14 +203,14 @@ async def _get(_: Request, resource_pool_id: int, user_id: str, user: base_model
             res = await self.repo.get_resource_pool_users(
                 keycloak_id=user_id, resource_pool_id=resource_pool_id, api_user=user
             )
-            if len(res) < 1:
-                raise errors.MissingResourceError(
-                    message=f"The user with id {user_id} or resource pool with id {resource_pool_id} cannot be found."
-                )
-            return json(
-                apispec.UserWithId(id=res[0].keycloak_id, no_default_access=res[0].no_default_access).model_dump(
-                    exclude_none=True
+            if len(res.allowed) > 0:
+                return json(
+                    apispec.UserWithId(
+                        id=res.allowed[0].keycloak_id, no_default_access=res.allowed[0].no_default_access
+                    ).model_dump(exclude_none=True)
```

Comment:
should there be an error that the resource pool does not allow any users?

Reply:
I am not sure what you mean by this? This is used for a request that tries to get a specific resource pool for a specific user. So if either there is an authorization problem or the user or the resource pool does not exist then we raise an error.

Reply:
But I see that for a default resource pool I was not adding the requesting users in the "allowed" list returned from the `get_resource_pool_users` properly. So I fixed that.

Reply:
i mean if `allowed` is less than 1 this would print `f"The user with id {user_id} or resource pool with id {resource_pool_id} cannot be found."` when actually the pool was found, it just doesn't allow any users

Reply:
As discussed this endpoint is to specifically check that a specific user can access a specific resource pool. If the permissions are not there then the requesting user should not know that the resource pool exists but they have no access to it.